chain/bitcoind_client_test.go

```diff
@@ -0,0 +1,126 @@
+)
+
+var (
+	NetParams = &chaincfg.RegressionNetParams
+)
+
+// NewMiner spawns testing harness backed by a bitcoind node that can serve as a
+// miner.
+func NewMiner(t *testing.T, extraArgs []string, createChain bool,
+	spendableOutputs uint32) (*rpctest.Harness, func()) {
+
+	t.Helper()
+
+	// Add the trickle interval argument to the extra args.
+	trickle := fmt.Sprintf("--trickleinterval=%v", TrickleInterval)
+	extraArgs = append(extraArgs, trickle)
+
+	node, err := rpctest.New(NetParams, nil, extraArgs, "")
+	require.NoError(t, err, "unable to create backend node")
+	if err := node.SetUp(createChain, spendableOutputs); err != nil {
+		node.TearDown()
+		t.Fatalf("unable to set up backend node: %v", err)
+	}
+
+	return node, func() { node.TearDown() }
+}
+
+// syncBitcoindWithMiner is a helper method that attempts to wait until the
+// bitcoind is synced (in terms of the chain) with the miner.
+func syncBitcoindWithMiner(t *testing.T, bitcoindClient *BitcoindClient,
+	miner *rpctest.Harness) uint32 {
+
+	t.Helper()
+
+	_, minerHeight, err := miner.Client.GetBestBlock()
+	require.NoError(t, err, "unable to retrieve miner's current height")
+
+	timeout := time.After(10 * time.Second)
+	for {
+		_, bitcoindHeight, err := bitcoindClient.GetBestBlock()
+		require.NoError(t, err, "unable to retrieve bitcoind's current "+
+			"height")
+
+		if bitcoindHeight == minerHeight {
+			return uint32(bitcoindHeight)
+		}
+
+		select {
+		case <-time.After(100 * time.Millisecond):
+		case <-timeout:
+			t.Fatalf("timed out waiting to sync bitcoind")
+		}
+	}
+}
+
+// TestBitcoindGetBlocksBatch ensures that we correctly retrieve
+// rawBlocks details using updated batchClient.
+func TestBitcoindGetBlocksBatch(t *testing.T) {
+	testBitcoindGetBlocksBatch(t, true)
+	testBitcoindGetBlocksBatch(t, false)
+}
+
+func testBitcoindGetBlocksBatch(t *testing.T, rpcpolling bool) {
+	miner, tearDown := NewMiner(
+		t, []string{"--txindex"}, true, 25,
+	)
+	defer tearDown()
+
+	bitcoindClient := setupBitcoind(t, miner.P2PAddress(), rpcpolling)
+
+	// Blocks shouldn't be retrieved from bitcoind when passing unknown
+	// block hashes.
+	var unknownHash1, unknownHash2, unknownHash3 chainhash.Hash
+	copy(unknownHash1[:], bytes.Repeat([]byte{0x10}, 32))
+	copy(unknownHash2[:], bytes.Repeat([]byte{0x12}, 32))
+	copy(unknownHash3[:], bytes.Repeat([]byte{0x14}, 32))
+	blockHashes := []*chainhash.Hash{&unknownHash1, &unknownHash2, &unknownHash3}
+	broadcastHeight := syncBitcoindWithMiner(t, bitcoindClient, miner)
+	_, err := bitcoindClient.GetBlocksBatch(blockHashes)
+	require.EqualError(t, err, "-5: Block not found")
+
+	// Here, we'll generate multiple valid testblocks and retrieve it
+	// linearly with GetBlock() (store the result for each GetBlock() in a
+	// slice) further also try to retrieve blocks in batch with GetBlocksBatch().
+	// At last will compare the results from both which should be equal.
+	if _, err := miner.Client.Generate(5); err != nil {
```

Comment:
nit: I think it'd be better to generate something like 2x15 + some blocks to test that we do create multiple batches correctly.